core/templates/pages/exploration-editor-page/services/change-list.service.ts

```diff
@@ -88,7 +91,20 @@ export class ChangeListService implements OnInit {
     private explorationDataService: ExplorationDataService,
     private loaderService: LoaderService,
     private loggerService: LoggerService,
-  ) {}
+    private connectionService: ConnectionService,
+  ) {
+    this.connectionService.onInternetStateChange.subscribe(
+      internetAccessible => {
+        if (internetAccessible) {
+          if (this.temporaryListOfChanges.length > 0) {
+            for (let change of this.temporaryListOfChanges) {
+              this.addChange(change);
```

Comment:
Yes we need this check because what addChange function gets called only when someone makes some changes. What if someone has already made changes when offline and then they were reconnected. Then we need to make calls to push the existing changes.